internal/resources/metal/organization/datasource.go

```diff
@@ -0,0 +1,79 @@
+package organization
+
+import (
+	"context"
+	"fmt"
+
+	equinix_errors "github.com/equinix/terraform-provider-equinix/internal/errors"
+	"github.com/equinix/terraform-provider-equinix/internal/framework"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/packethost/packngo"
+)
+
+func NewDataSource() datasource.DataSource {
+	return &DataSource{
+		BaseDataSource: framework.NewBaseDataSource(
+			framework.BaseDataSourceConfig{
+				Name: "equinix_metal_organization",
+			},
+		),
+	}
+}
+
+type DataSource struct {
+	framework.BaseDataSource
+}
+
+func (r *DataSource) Schema(
+	ctx context.Context,
+	req datasource.SchemaRequest,
+	resp *datasource.SchemaResponse,
+) {
+	resp.Schema = dataSourceSchema(ctx)
+}
+
+func (r *DataSource) Read(
+	ctx context.Context,
+	req datasource.ReadRequest,
+	resp *datasource.ReadResponse,
+) {
+	r.Meta.AddFwModuleToMetalUserAgent(ctx, req.ProviderMeta)
+	client := r.Meta.Metal
+
+	// Retrieve values from plan
+	var data DataSourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Extract the ID of the resource from the state
+	id := data.OrganizationID.ValueString()
+
+	var org *packngo.Organization
+
+	// orgs, _, err := client.Organizations.List(&packngo.GetOptions{Includes: []string{"address"}})
```

Comment:
Continuing thread from https://github.com/equinix/terraform-provider-equinix/pull/568#discussion_r1499500723:

This is still not equivalent to the SDKv2 datasource.  The datasource should accept _either_ a name or an ID.  If it is given an ID, it should run a `Get`.  If it is given a name, it should run a `List` and then look through the list for an organization with a matching name.